core/src/main/java/io/grpc/util/GracefulSwitchLoadBalancer.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.util;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.grpc.ConnectivityState.READY;
+
+import io.grpc.ConnectivityState;
+import io.grpc.Internal;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancerProvider;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A forwarding load balancer and holder of currentLb and pendingLb. The pendingLb's helper will not
+ * update balancing state until a subchannel managed by the pendingLB is READY, whence the pendingLb
+ * becomes current.
+ */
+@Internal
```

Comment:
This is because `ForwardingLoadBalancer` and `ForwardingLoadBalancerHelper` are in `io.grpc.util`. We had the same problem for `CachedSubchannelPool` https://github.com/grpc/grpc-java/pull/5663#discussion_r280648764

Reply:
We should either move the Forwarding tools to `io.grpc` or to `io.grpc.internal`. We can have a shim in `io.grpc.util` that extends from its "real" location.

Reply:
In talking with Kun more:
 * This is a utility that hierarchical LBs will need. Long-term it should be a public utility (in io.grpc.util)
 * We will need to use it in `io.grpc.internal` for AutoConfiguredLoadBalancer

So that means the core implementation will need to be in `io.grpc.internal` and then we'll have a shim in `io.grpc.util`. The shim can be a ForwardingLoadBalancer where delegate() returns the instance from `io.grpc.internal` and defines switchTo().

So this current location is "fine", but shouldn't be `@Internal`.

Reply:
I had following choices in mind (out of scope of this PR):
- A) Move Forwarding tools from util to `io.grpc`. This need deprecate the existing API.
- B) Move Forwarding tools from util to `io.grpc.internal`. This need deprecate the existing API, and  after deleted there will be no external public API anymore.
- C) Add an InternalAccessor shim in `io.grpc.util`. Then `io.grpc.util` will be more dependent on `io.grpc.internal`. (Currently `io.grpc.util` does depend on `io.grpc.internal`, but technically not necessary).
- D) Let `io.grpc.internal` depend on `io.grpc.util`, not currently the opposite. Will impact Bazel users.
- E) Move RoundRobinBalancer out of `io.grpc.util` to `io.grpc.roundrobin` (this will make `io.grpc.util` lightweight), and let `io.grpc.internal` depend on the lightweight `io.grpc.util`, not the opposite. This will have behavior change: 
  + people using round_robin will add `io.grpc.roundrobin` to runtime dependency; bazel users using round_robin will not add `io.grpc.util` to runtime dependency, which currently they do and does not make sense.
  + those who only want to use the public APIs in `io.grpc.util` will not have a side effect of loading round_robin.